Formula/p/pgvector.rb

```diff
@@ -16,35 +17,43 @@ class Pgvector < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "a0724750eaedf54c546ccc75dae7613cef15f773bb189eb531bd5bba00f1732a"
   end
 
-  depends_on "postgresql@16"
+  depends_on "postgresql@15" => [:build, :test]
+  depends_on "postgresql@16" => [:build, :test]
 
-  def postgresql
-    Formula["postgresql@16"]
+  def postgresqls
+    deps.map(&:to_formula).sort_by(&:version).filter { |f| f.name.start_with?("postgresql@") }
   end
 
   def install
-    ENV.prepend_path "PATH", postgresql.opt_libexec/"bin"
-    system "make"
-    system "make", "install", "pkglibdir=#{lib/postgresql.name}",
-                              "datadir=#{share/postgresql.name}",
-                              "pkgincludedir=#{include/postgresql.name}"
+    postgresqls.each do |postgresql|
+      ENV.prepend_path "PATH", postgresql.opt_libexec/"bin"
```

Comment:
Can also use `ENV["PG_CONFIG"] = postgresql.opt_libexec/"bin/pg_config"` to avoid modifying the path.